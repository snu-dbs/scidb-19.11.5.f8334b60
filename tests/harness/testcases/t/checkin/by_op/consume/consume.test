--setup
create array Xa <a:int64> [ X=0:9,5,0]
create array Xb <b:int64> [ X=0:9,5,0]
create array Xc <c:int64> [ X=0:9,5,0]
create array Xd <d:int64> [ X=0:9,5,0]

store(build(Xa, 1+X), Xa)
scan(Xa)  -- Inserted by scan_doctor
store(build(Xb, 1+X), Xb)
scan(Xb)  -- Inserted by scan_doctor
store(build(Xc, 1+X), Xc)
scan(Xc)  -- Inserted by scan_doctor
store(build(Xd, 1+X), Xd)
scan(Xd)  -- Inserted by scan_doctor

store(join(Xa, Xb), X1)
scan(X1)  -- Inserted by scan_doctor
store(join(X1, Xc), X2)
scan(X2)  -- Inserted by scan_doctor
store(join(X2, Xd), X3)
scan(X3)  -- Inserted by scan_doctor

create array Ya <a:int64>[x=0:*,5,0]
create array Yb <b:int64>[x=0:*,5,0]
create array Yc <c:int64>[x=0:*,5,0]
create array Yd <d:int64>[x=0:*,5,0]

load(Ya, '${TEST_DATA_DIR}/unbounded_cross.txt')
scan(Ya)  -- Inserted by scan_doctor
load(Yb, '${TEST_DATA_DIR}/unbounded_cross.txt')
scan(Yb)  -- Inserted by scan_doctor
load(Yc, '${TEST_DATA_DIR}/unbounded_cross.txt')
scan(Yc)  -- Inserted by scan_doctor
load(Yd, '${TEST_DATA_DIR}/unbounded_cross.txt')
scan(Yd)  -- Inserted by scan_doctor

store(join(Ya, Yb), Y1)
scan(Y1)  -- Inserted by scan_doctor
store(join(Y1, Yc), Y2)
scan(Y2)  -- Inserted by scan_doctor
store(join(Y2, Yd), Y3)
scan(Y3)  -- Inserted by scan_doctor

--test

consume(merge(Y3,X3))
consume(merge(Y3,X3),2)
consume(merge(Y3,X3),4)
consume(merge(Y3,X3),-1)
consume(merge(Y3,X3),0)
--error --code=scidb::SCIDB_SE_SYNTAX::SCIDB_LE_WRONG_OPERATOR_ARGUMENTS_COUNT --afl="consume(merge(Y3,X3),4,5)"
--error --code=scidb::SCIDB_SE_TYPE::SCIDB_LE_CANT_FIND_IMPLICIT_CONVERTER --afl="consume(merge(Y3,X3),4.5)"

--cleanup
remove(Xa)
remove(Xb)
remove(Xc)
remove(Xd)
remove(X1)
remove(X2)
remove(X3)
remove(Ya)
remove(Yb)
remove(Yc)
remove(Yd)
remove(Y1)
remove(Y2)
remove(Y3)
