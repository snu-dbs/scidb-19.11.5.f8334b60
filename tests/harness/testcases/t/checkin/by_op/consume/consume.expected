Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

{X} a
{0} 1
{1} 2
{2} 3
{3} 4
{4} 5
{5} 6
{6} 7
{7} 8
{8} 9
{9} 10

Query was executed successfully

{X} b
{0} 1
{1} 2
{2} 3
{3} 4
{4} 5
{5} 6
{6} 7
{7} 8
{8} 9
{9} 10

Query was executed successfully

{X} c
{0} 1
{1} 2
{2} 3
{3} 4
{4} 5
{5} 6
{6} 7
{7} 8
{8} 9
{9} 10

Query was executed successfully

{X} d
{0} 1
{1} 2
{2} 3
{3} 4
{4} 5
{5} 6
{6} 7
{7} 8
{8} 9
{9} 10

Query was executed successfully

{X} a,b
{0} 1,1
{1} 2,2
{2} 3,3
{3} 4,4
{4} 5,5
{5} 6,6
{6} 7,7
{7} 8,8
{8} 9,9
{9} 10,10

Query was executed successfully

{X} a,b,c
{0} 1,1,1
{1} 2,2,2
{2} 3,3,3
{3} 4,4,4
{4} 5,5,5
{5} 6,6,6
{6} 7,7,7
{7} 8,8,8
{8} 9,9,9
{9} 10,10,10

Query was executed successfully

{X} a,b,c,d
{0} 1,1,1,1
{1} 2,2,2,2
{2} 3,3,3,3
{3} 4,4,4,4
{4} 5,5,5,5
{5} 6,6,6,6
{6} 7,7,7,7
{7} 8,8,8,8
{8} 9,9,9,9
{9} 10,10,10,10

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

{x} a
{0} 0
{2} 1
{4} 2
{6} 3
{8} 4
{10} 5
{12} 6
{14} 7
{16} 8
{18} 9
{20} 10
{22} 11
{24} 12
{26} 13
{28} 14
{29} 15

Query was executed successfully

{x} b
{0} 0
{2} 1
{4} 2
{6} 3
{8} 4
{10} 5
{12} 6
{14} 7
{16} 8
{18} 9
{20} 10
{22} 11
{24} 12
{26} 13
{28} 14
{29} 15

Query was executed successfully

{x} c
{0} 0
{2} 1
{4} 2
{6} 3
{8} 4
{10} 5
{12} 6
{14} 7
{16} 8
{18} 9
{20} 10
{22} 11
{24} 12
{26} 13
{28} 14
{29} 15

Query was executed successfully

{x} d
{0} 0
{2} 1
{4} 2
{6} 3
{8} 4
{10} 5
{12} 6
{14} 7
{16} 8
{18} 9
{20} 10
{22} 11
{24} 12
{26} 13
{28} 14
{29} 15

Query was executed successfully

{x} a,b
{0} 0,0
{2} 1,1
{4} 2,2
{6} 3,3
{8} 4,4
{10} 5,5
{12} 6,6
{14} 7,7
{16} 8,8
{18} 9,9
{20} 10,10
{22} 11,11
{24} 12,12
{26} 13,13
{28} 14,14
{29} 15,15

Query was executed successfully

{x} a,b,c
{0} 0,0,0
{2} 1,1,1
{4} 2,2,2
{6} 3,3,3
{8} 4,4,4
{10} 5,5,5
{12} 6,6,6
{14} 7,7,7
{16} 8,8,8
{18} 9,9,9
{20} 10,10,10
{22} 11,11,11
{24} 12,12,12
{26} 13,13,13
{28} 14,14,14
{29} 15,15,15

Query was executed successfully

{x} a,b,c,d
{0} 0,0,0,0
{2} 1,1,1,1
{4} 2,2,2,2
{6} 3,3,3,3
{8} 4,4,4,4
{10} 5,5,5,5
{12} 6,6,6,6
{14} 7,7,7,7
{16} 8,8,8,8
{18} 9,9,9,9
{20} 10,10,10,10
{22} 11,11,11,11
{24} 12,12,12,12
{26} 13,13,13,13
{28} 14,14,14,14
{29} 15,15,15,15

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

[An error expected at this place for the query "consume(merge(Y3,X3),4,5)". And it failed with error code = scidb::SCIDB_SE_SYNTAX::SCIDB_LE_WRONG_OPERATOR_ARGUMENTS_COUNT. Expected error code = scidb::SCIDB_SE_SYNTAX::SCIDB_LE_WRONG_OPERATOR_ARGUMENTS_COUNT.]

[An error expected at this place for the query "consume(merge(Y3,X3),4.5)". And it failed with error code = scidb::SCIDB_SE_TYPE::SCIDB_LE_CANT_FIND_IMPLICIT_CONVERTER. Expected error code = scidb::SCIDB_SE_TYPE::SCIDB_LE_CANT_FIND_IMPLICIT_CONVERTER.]

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

